Assets/Treasure/TDK/Runtime/API/Magicswap.cs

```diff
@@ -189,9 +238,18 @@ public async Task<MagicswapRoute> GetRoute(string tokenInId, string tokenOutId,
 
         public async Task<Transaction> Swap(SwapBody swapBody) {
             swapBody.backendWallet ??= TDK.AppConfig.GetBackendWallet();
+            swapBody.toAddress ??= TDK.Identity.Address;
             var body = JsonConvert.SerializeObject(swapBody, new JsonSerializerSettings {
                 NullValueHandling = NullValueHandling.Ignore
             });
+            var thirdwebService = TDKServiceLocator.GetService<TDKThirdwebService>();
+            if (await thirdwebService.IsZkSyncChain(TDK.Connect.ChainIdNumber))
```

Comment:
Can this be stored in the `TDK.Connect` class when chain is switched? Don't want to make a network request to Thirdweb's API every time this is checked

Reply:
This is already cached internally by thirdweb, so it would only be fetched once at most
<img width="812" alt="image" src="https://github.com/user-attachments/assets/47abbd9d-1fa7-4eb1-a347-164a19e18f36">